### ISSUE TITLE
issue in joint_test.R 

### ISSUE THREAD

**Issue by zkx1138** [immunogenomics/GeNA#4]
Hey nice work on GeNA. 

We noticed a potential bug in the following line:
https://github.com/immunogenomics/GeNA/blob/cc41623bf8b42b819caa6b3024d777476a8adf10/joint_test.R#L26

If the earlier call to pchisq returns NA's for all of the k[ks]_P values in that row of the matrix (due to a numeric issue?) the apply becomes really slow and casts the vector into a list. Can you catch this case or fix? We've filtered variants on MAF and missigness so that's not the cause. Not sure why this is happening. 

Thanks!

**Comment by rumker** [comment]
Thank you for reaching out! We’ll look into this and follow up soon.

**Comment by rumker** [comment]
Hi @zia1138, I'm sorry it's taken me a while to get back to you and thank you for your patience!

I suspect the chi-square value for one of the NAM-PCs less than or equal to the smallest considered k value (e.g., NAM-PC1, NAM-PC2, or NAM-PC3) is an NaN and I'm curious why that might be happening. Would you be willing to share the following with me for your problematic SNP:
- What the row of `all_res` corresponding to that SNP looks like as soon as it's imported to joint_test.R?
- What k values are being used? 
- If there is an NaN in the imported `all_res` where I suspect there may be (let's say it's NAM-PC3), what the row of plink_per_nampc/NAM.PC3.glm.linear corresponding to that SNP looks like?

Thank you!

**Comment by zkx1138** [comment]
Thanks @rumker. We found out it was due to the order we were filtering variants and individuals in the cohort. We ended up with variants with MAF of 0 or too many NAs. This led to all NA's in the glm.linear files for that variant. All it takes is one variant with NAs in all the glm.linear file for the script to fail. It might be good to add a check for NAs before line 26 as R just casts the object into a list and everything slows down and memory gets eaten up. Thanks again for your hard work on GeNA!
